validator/tests/test_scheduler/tests.py

```diff
@@ -162,10 +213,158 @@ def test_serial_batch_fails_valid_txn(self):
+            scheduler=scheduler,
+            context_manager=context_manager,
+            name='complex_batches_multiple_failures.yaml')
+
+    def test_parallel_enclosing_writer_fails(self):
+        """Tests the parallel scheduler against the
+        test_scheduler/data/enclosing_writer_fails.yaml file.
+
+        Notes:
+            This yaml file has failed batches that have txns that implicitly
+            depend on multiple prior txns because they have the
+            whole tree ('') in the inputs and outputs, and also are
+            the implicit dependency of several subsequent txns. When
+            the batch fails, those dependent txns need to be replayed with
+            a different state.
+        """
+
+        context_manager, scheduler = self._setup_parallel_scheduler()
+        self._single_block_files_individually(
+            scheduler=scheduler,
+            context_manager=context_manager,
+            name='enclosing_writer_fails.yaml')
+
+    def test_parallel_lifo_enclosing_writer_fails(self):
+        """Tests the parallel scheduler against the
+        test_scheduler/data/enclosing_writer_fails.yaml file.
+
+        Notes:
+            This yaml file has failed batches that have txns that implicitly
+            depend on multiple prior txns because they have the
+            whole tree ('') in the inputs and outputs, and also are
+            the implicit dependency of several subsequent txns. When
+            the batch fails, those dependent txns need to be replayed with
+            a different state.
+        """
+
+        context_manager, scheduler = self._setup_parallel_scheduler()
+        self._single_block_files_individually(
+            scheduler=scheduler,
+            context_manager=context_manager,
+            name='enclosing_writer_fails.yaml',
+            lifo=True)
+
+    def test_serial_enclosing_writer_fails(self):
+        """Tests the serial scheduler against the
+        test_scheduler/data/complex_batches_multiple_failures.yaml file.
+
+        Notes:
+            This yaml file has failed batches that have txns that implicitly
+            depend on multiple prior txns because they have the
+            whole tree ('') in the inputs and outputs, and also are
+            the implicit dependency of several subsequent txns. When
+            the batch fails, those dependent txns need to be replayed with
+            a different state.
+        """
+
+        context_manager, scheduler = self._setup_serial_scheduler()
+        self._single_block_files_individually(
+            scheduler=scheduler,
+            context_manager=context_manager,
+            name='enclosing_writer_fails.yaml')
+
+    def test_parallel_heterogeneous_workload(self):
+        """Tests the parallel scheduler against the
+        test_scheduler/data/heterogeneous_workload.yaml file.
+
+        Notes:
+            This yaml file has namespaced addresses and several failed batches
+        """
```

Comment:
Please add some more description of what this test is doing (along with the subsequent tests)